src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -615,6 +649,102 @@ const getSshInfo = async (frameworkName) => {
   throw createError('Not Found', 'NoJobSshInfoError', `SSH info of job ${frameworkName} is not found.`);
 };
 
+const extractDiagnosticsInfo = (diag) => {
+  if (_.isEmpty(diag)) {
+    return null;
+  }
+  // The pattern for diagnostics is "podPattern unmatched: json" or "podPattern nmatched: json"
+  const regex = /matched: (.*)/;
+  const matches = diag.match(regex);
+  if (matches.length < 2) {
+    return null;
+  }
+  try {
+    return JSON.parse(matches[1]);
+  } catch (error) {
+    logger.warn('Get diagnostics info failed', error);
+  }
+  return null;
+};
+
+const extractRuntimeOutput = (diag) => {
+  if (_.isEmpty(diag)) {
+    return null;
+  }
+
+  // Do not container user container info
+  if (_.isNil(diag.containers) || diag.containers.length < 2) {
+    return null;
+  }
+  const message = diag.containers[1].message;
+
+  const anchor1 = /\[PAI_RUNTIME_ERROR_START\]/;
+  const anchor2 = /\[PAI_RUNTIME_ERROR_END\]/;
+  const match1 = message.match(anchor1);
+  const match2 = message.match(anchor2);
+  if (match1 !== null && match2 !== null) {
+    const start = match1.index + match1[0].length;
+    const end = match2.index;
+    const output = message.substring(start, end).trim();
+    return yaml.safeLoad(output);
+  }
+};
+
+const extractContainerOutput = (diag) => {
+  if (_.isEmpty(diag) || _.isEmpty(diag.containers)) {
+    return null;
+  }
+  const anchor = /\[PAI_RUNTIME_ERROR_START\]/;
+
+  const obj = {};
+  diag.containers.forEach((container) => {
+    let message = container.message;
+    if (_.isEmpty(message)) {
+      return;
+    }
+    const match = message.match(anchor);
+    if (match !== null) {
+      message = message.substring(0, match.index);
+    }
+    if (_.isEmpty(message)) {
+      return;
+    }
+    obj[container.name] = message;
+  });
+  if (_.isEmpty(obj)) {
+    return null;
+  }
+  return obj;
+};
+
+const extractLauncherOutput = (diag, code) => {
+  if (_.isEmpty(diag) || code > 0) {
+    return null;
+  }
+  return diag;
```

Comment:
>diag [](start = 9, length = 4)

return this, when PAI_RUNTIME_ERROR_START cannot be found, instead of  code comparing with 0.
Such as when SIGKILL received